transform-plugins/docs/Flatten-transform.md

```diff
@@ -0,0 +1,37 @@
+# Flatten Transform
+
+## Description
+Flatten is a transform plugin that flattens nested data structures.
+
+## Use Case
+
+The transform is used  to convert a nested data structure  into a single flattened record where each key in the record 
+is a ```_``` name path to the node in the nested structure. 
+
+Configuration
+-------------
+**Fields to flatten:** Specifies the list of fields in the input schema to be flattened.
+
+**Level to limit flattening:** Limit flattening to a certain level in nested structures.
```

Comment:
why is this needed? Cant we just rely on structured record to figure out nesting levels?

Reply:
@CuriousVini 
As for the requirments we can restricts nesting to certain level, where the default is 1 level.

Reply:
I see